front/gatsby/src/components/Layout/Header/Header.tsx

```diff
@@ -2,91 +2,78 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { useSubscription } from '@apollo/react-hooks';
-import { AccountsContext } from '@substrate/context';
+import { useQuery, useSubscription } from '@apollo/react-hooks';
+import { formatBalance } from '@polkadot/util';
+import { AccountsContext, ApiContext } from '@substrate/context';
 import { Button, ItemStats } from '@substrate/design-system';
-import gql from 'graphql-tag';
-import React, { useContext, useEffect, useState } from 'react';
+import { Container, Grid } from '@substrate/ui-components';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
 
 import { APP_TITLE, toShortAddress } from '../../../util';
+import {
+  BLOCKS_SUBSCRIPTION,
+  ERAS_SUBSCRIPTION,
+  LATEST_ERA_QUERY,
+  LATEST_SESSION_QUERY,
+  SESSIONS_SUBSCRIPTION,
+  STAKING_SUBSCRIPTION,
+} from '../graphql';
 import styles from './Header.module.css';
 import { BlockHead, EraHead, SessionHead, StakingHead } from './types';
 
-const BLOCKS_SUBSCRIPTION = gql`
-  subscription {
-    subscribeBlockNumbers {
-      authoredBy
-      hash
-      number
-      startDateTime
-    }
-  }
-`;
-
-const ERAS_SUBSCRIPTION = gql`
-  subscription {
-    subscribeEras {
-      index
-      totalPoints
-    }
-  }
-`;
-
-const SESSIONS_SUBSCRIPTION = gql`
-  subscription {
-    subscribeSessions {
-      index
-    }
-  }
-`;
-
-const STAKING_SUBSCRIPTION = gql`
-  subscription {
-    subscribeStakes {
-      blockNumber {
-        number
-      }
-      totalStake
-    }
-  }
-`;
-
-const EraHeader = () => {
+const EraHeader = (): React.ReactElement => {
+  const { api } = useContext(ApiContext);
   const { data } = useSubscription(ERAS_SUBSCRIPTION);
+  const queryData = useQuery(LATEST_ERA_QUERY);
```

Comment:
I'm curious: why do we need both?

Reply:
the subscription only listens for a mutation, i.e. a new era was created. So if we just subscribe, this field will be empty till the next era is created. 

An alternative way to the above is to use polling queries but I think for something that doesn't change very often such as eras, the current implementation makes more sense.